crates/autopilot/src/run_loop.rs

```diff
@@ -211,20 +226,32 @@ impl RunLoop {
                 return;
             }
 
-            tracing::info!(driver = %driver.name, "settling");
-            let submission_start = Instant::now();
-            match self
-                .settle(driver, solution, auction_id, block_deadline)
-                .await
-            {
-                Ok(()) => Metrics::settle_ok(driver, submission_start.elapsed()),
-                Err(err) => {
-                    Metrics::settle_err(driver, submission_start.elapsed(), &err);
-                    tracing::warn!(?err, driver = %driver.name, "settlement failed");
+            let solution_id = solution.id();
+            let solved_order_uids = solution.orders().keys().cloned().collect();
```

Comment:
`solved_order_uids` is only needed for adding them to the cache which should probably happen before spawning the task just to be safe and avoid possible race conditions.